network-config-analyzer/ResourcesHandler.py

```diff
@@ -0,0 +1,223 @@
+
+        if len(self.pods_finder.peer_set) > 0 and len(self.ns_finder.namespaces) > 0:
+            return True, 0
+        elif len(self.pods_finder.peer_set) == 0 and len(self.ns_finder.namespaces) > 0:
+            return False, ResourceType.Namespaces
+        elif len(self.pods_finder.peer_set) > 0 and len(self.ns_finder.namespaces) == 0:
+            return False, ResourceType.Pods
+        else:
+            return False, 0
+
+    def parse_lists_for_policies(self, np_list, resource_list, peer_container):
+        """
+        parses policies from np_list resource if exists, otherwise parses the resource_list for policies
+        :param np_list: list of entries of networkPolicies
+        :param resource_list: list of entries
+        :param peer_container: the existing PeerContainer
+        """
+        self.policies_finder.set_peer_container(peer_container)
+        if np_list and np_list != ['']:
+            self._parse_resources_path(np_list, [ResourceType.Policies])
+            if resource_list:
+                print('Warning: policies will be taken from the provided networkPolicy list key. '
+                      'input resources provided with resource list key will be ignored when finding network policies')
+        elif resource_list:
+            self._parse_resources_path(resource_list, [ResourceType.Policies])
+
+    def _parse_resources_path(self, resource_list, resource_flags):
+        """
+        parsing the resources path / live cluster using the Finder classes
+        """
+        for resource_item in resource_list:
+            if resource_item == 'k8s':
+                self.load_resources_from_k8s_live_cluster(resource_flags)
+            elif resource_item == 'calico' and ResourceType.Pods in resource_flags:
+                self.pods_finder.load_peer_from_calico_resource()
+            elif resource_item == 'calico' and ResourceType.Policies in resource_flags:
+                self.policies_finder.load_policies_from_calico_cluster()
+            elif resource_item == 'istio' and ResourceType.Policies in resource_flags:
+                self.policies_finder.load_istio_policies_from_k8s_cluster()
+            else:
+                resource_scanner = TreeScannerFactory.get_scanner(resource_item)
+                if resource_scanner is None:
+                    continue
+                yaml_files = resource_scanner.get_yamls()
+                if not yaml_files:
+                    continue
+                for yaml_file in yaml_files:
+                    for res_code in yaml_file.data:
+                        if ResourceType.Namespaces in resource_flags:
+                            self.ns_finder.parse_yaml_code_for_ns(res_code)
+                        if ResourceType.Pods in resource_flags:
+                            self.pods_finder.namespaces_finder = self.ns_finder
+                            self.pods_finder.add_eps_from_yaml(res_code)
```

Comment:
In recent commit to the master branch, instead of sending the `res_code` to this function (`add_eps_from_yaml`), the `yaml_file.data` is sent.
I kept it with sending the res_code, since:
1. I think it works the same with res_code and yaml_file.data (I made all the changes in the function itself to match that commit's changes)
2. yaml_file.data is a generator , sending it as a parameter (to the functions) will cause tests with the new switch "resource_list" to fail (e.g for tests with resource_list that replaces ns_list and pod_list, the generator will find only namespaces)

So, I would like to know if keeping the `res_code` is okay for you. (is there any case that may fail if we don't send the `yaml_file.data`)

Reply:
From a short conversation with Ziv, this is good too